marketing/v19/adset.go

```diff
@@ -100,6 +102,78 @@ func (as *AdsetService) Update(ctx context.Context, a Adset) (fb.Time, error) {
 	return res.UpdatedTime, nil
 }
 
+// Delete deletes an adset.
+func (as *AdsetService) Delete(ctx context.Context, id string) error {
+	res := &fb.MinimalResponse{}
+	err := as.c.DeleteJSON(ctx, fb.NewRoute(Version, "/%s", id).String(), nil, res)
+	if err != nil {
+		return err
+	} else if err = res.GetError(); err != nil {
+		return err
+	} else if !res.Success {
+		return fmt.Errorf("deleting the adset failed")
+	}
+
+	return nil
+}
+
+// CopyAsync copies an adset using async batch.
+func (as *AdsetService) CopyAsync(ctx context.Context, id string) (*fb.CopiedAdsetAsyncBatchResult, error) {
+	if id == "" {
+		return nil, errors.New("cannot copy adset without id")
+	}
+
+	res := &fb.AsyncBatchCreateResponse{}
+
+	req := &fb.AsyncBatchCreateRequest{
+		AsyncBatch: []fb.AsyncBatchOperation{
+			{
+				Method:      "POST",
+				RelativeURL: fmt.Sprintf("%s/copies", id),
+				Name:        fmt.Sprintf("%s_copy", id),
+				Body:        "deep_copy=true",
+			},
+		},
+	}
+
+	err := as.c.PostJSON(ctx, fb.NewRoute(Version, "/").String(), req, res)
+	if err != nil {
+		return nil, err
+	}
+
+	asyncSessionID := res.AsyncSessions[0].ID
+
+	var result *fb.CopiedAdsetAsyncBatchResult
+
+	for {
+		asyncBatchStatus := &fb.AsyncBatch{}
+
+		err = as.c.GetJSON(ctx, fb.NewRoute(Version, "/%s", asyncSessionID).
+			Fields("result", "status", "error_code", "exception").
```

Comment:
fb.AsyncBatch has much more fields than we get from FB. Do we need them?
If not could you remove them from struct to avoid additional questions